core/src/main/java/org/apache/iceberg/BaseSnapshot.java

```diff
@@ -64,6 +68,9 @@ class BaseSnapshot implements Snapshot {
             path -> new GenericManifestFile(io.newInputFile(path), 0)));
   }
 
+  // Use the version that doesn't take in FileIO - calls for FileIO should be handled by the caller for when
+  // serialized TableMetadata is parsed (without a FileIO to push down).
```

Comment:
We typically add this information with `@deprecated` in Javadoc, along with when it will be removed. (1.0.0)

Reply:
Yeah our checkstyle caught it.